src/cdk-experimental/menu/menu-item-trigger.spec.ts

```diff
@@ -38,20 +40,288 @@ describe('MenuItemTrigger', () => {
+      triggers[0].toggle();
+      detectChanges();
+
+      expect(menus.length).toEqual(1);
+      expect(triggers[0]._menuPanel!._menu!).toEqual(menus[0]);
+    });
+
+    it('should not open the submenu when menu item disabled', () => {
+      menuItems[0].disabled = true;
+
+      menuItems[0].trigger();
+      detectChanges();
+
+      expect(menus.length).toBe(0);
+    });
+
+    it('should toggle the attached submenu', () => {
+      triggers[0].toggle();
+      detectChanges();
+      expect(menus.length).toEqual(1);
+
+      triggers[0].toggle();
+      detectChanges();
+      expect(menus.length).toEqual(0);
+    });
+
+    it('should open a nested submenu when nested trigger is clicked', () => {
+      triggers[0].toggle();
+      detectChanges();
+
+      triggers[1].toggle();
+      detectChanges();
+
+      expect(menus.length).toEqual(2);
+    });
+
+    it('should close all sub-menus when root menu is closed', () => {
+      triggers[0].toggle();
+      detectChanges();
+      triggers[1].toggle();
+      detectChanges();
+
+      expect(menus.length).toEqual(2);
+
+      triggers[0].toggle();
+      detectChanges();
+
+      expect(menus.length).toEqual(0);
+    });
+
+    it('should close nested submenu and leave parent open', () => {
+      triggers[0].toggle();
+      detectChanges();
+      triggers[1].toggle();
+      detectChanges();
+
+      expect(menus.length).toEqual(2);
+
+      triggers[1].toggle();
+      detectChanges();
+
+      expect(menus.length).withContext('first level menu should stay open').toEqual(1);
+    });
+
+    it('should emit request to open event on submenu open', () => {
+      const triggerSpy = jasmine.createSpy('cdkMenuItem open request emitter');
+      triggers[0].opened.subscribe(triggerSpy);
+
+      triggers[0].toggle();
+
+      expect(triggerSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should emit request to close event on submenu close', () => {
+      const triggerSpy = jasmine.createSpy('cdkMeuItem close request emitter');
+      const closedSpy = jasmine.createSpy('cdkMenu closed emitter');
+      triggers[0].closed.subscribe(triggerSpy);
+
+      triggers[0].toggle();
+      detectChanges();
+      menus[0].closed.subscribe(closedSpy);
+
+      triggers[0].toggle();
+      detectChanges();
+
+      expect(triggerSpy).toHaveBeenCalledTimes(1);
+      expect(closedSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should position the overlay below the trigger by default for a horizontal Menu', () => {
+      triggers[0].toggle();
+      detectChanges();
+
+      expect(Math.floor(nativeTriggers[0].getBoundingClientRect().bottom))
+        .withContext('MenuBar is horizontal by default')
+        .toEqual(Math.floor(nativeMenus[0].getBoundingClientRect().top));
+    });
+
+    it('should fallback to positioning the overlay above the trigger for horizontal Menu', () => {
+      nativeTriggers[0].style.position = 'fixed';
+      nativeTriggers[0].style.bottom = '0';
```

Comment:
Since the position is set to fixed, setting the bottom to 0 sets it 0px away from the bottom edge. Similar logic applies for `right` below.

Reference https://developer.mozilla.org/en-US/docs/Web/CSS/bottom and https://developer.mozilla.org/en-US/docs/Web/CSS/right

-----

I'll clear up the comment though

Reply:
Wow, I had no idea that was a thing. CSS is a mystery...